server/src/main/java/io/crate/expression/scalar/ArrayUpperFunction.java

```diff
@@ -210,13 +210,13 @@ public Query toQuery(Function parent, Function arrayLength, Context context) {
 
             case GtOperator.NAME:
                 if (cmpVal == 0) {
-                    return IsNullPredicate.refExistsQuery(arrayRef, context, false);
+                    return new FieldExistsQuery(arrayRef.storageIdent());
```

Comment:
This does not affect functionality just refactoring in order to reduce the complexity of `IsNullPredicate.refExistsQuery`.

Reply:
Checking `IsNullPredicate.refExistsQuery`, it has quite some logic and `FieldExistsQuery` is used only under specific conditions. 

Even after excluding `if (countEmptyArrays)`  block, we check there doc values being enabled, index type (see `canUseFieldsExist` flag).

Are we sure this doesn't affect functionality? Have you tested `array(geo_shape)`?

Reply:
We return `null` for geoshape arrays,
```
        if (elementType.id() == ObjectType.ID || elementType.equals(DataTypes.GEO_SHAPE)) {
            // No doc-values for these, can't utilize doc-value-count
            return null;
        }
```

I will try to find a test or add one to make sure we can use `FieldExistsQuery`

Reply:
Please check `test_is_null_on_arrays` this test passes on `master` as well. Had this test but some reason  I deleted...

Reply:
Thanks for catching but please ignore this, I have dropped the commit. I will handle it in the other PR if anything.